src/main/java/com/github/luben/zstd/ZstdInputStreamNoFinalizer.java

```diff
@@ -146,21 +146,20 @@ int readInternal(byte[] dst, int offset, int len) throws IOException {
                     + " from offset " + offset + " in buffer of size " + dst.length);
         }
         int dstSize = offset + len;
-        dstPos = offset;
```

Comment:
You cannot do this - the `dstPos` (and also `srcPos`) are modified by the native code.